### ISSUE TITLE
Reproduction issue for SYSU

### ISSUE THREAD

**Issue by 4fouroy4** [mmunhappy/ICASSP2025-PDM#5]
Hi,

I hope this message finds you well.

First of all, thank you for your work and for sharing the code. I've run the training using the command:
 **python train_pdm_os.py --dataset sysu --gpu 0 --protos 10**

The best results were as follows:

Epoch: [78][0/695] DCSLoss:0.184 DPLoss:-0.225 Loss:0.768 iLoss:0.077 TLoss:0.159 CLoss:0.904 OLoss:0.010 
Epoch: [78][50/695] DCSLoss:0.222 DPLoss:-0.226 Loss:1.364 iLoss:0.536 TLoss:0.248 CLoss:0.921 OLoss:0.009 
Epoch: [78][100/695] DCSLoss:0.183 DPLoss:-0.226 Loss:0.821 iLoss:0.110 TLoss:0.168 CLoss:0.926 OLoss:0.010 
Epoch: [78][150/695] DCSLoss:0.246 DPLoss:-0.225 Loss:1.297 iLoss:0.342 TLoss:0.266 CLoss:1.091 OLoss:0.010 
Epoch: [78][200/695] DCSLoss:0.181 DPLoss:-0.226 Loss:1.016 iLoss:0.329 TLoss:0.166 CLoss:0.887 OLoss:0.010 
Epoch: [78][250/695] DCSLoss:0.196 DPLoss:-0.224 Loss:1.106 iLoss:0.340 TLoss:0.201 CLoss:0.943 OLoss:0.010 
Epoch: [78][300/695] DCSLoss:0.207 DPLoss:-0.224 Loss:1.308 iLoss:0.497 TLoss:0.219 CLoss:0.973 OLoss:0.009 
Epoch: [78][350/695] DCSLoss:0.174 DPLoss:-0.224 Loss:0.940 iLoss:0.257 TLoss:0.166 CLoss:0.891 OLoss:0.010 
Epoch: [78][400/695] DCSLoss:0.209 DPLoss:-0.226 Loss:1.055 iLoss:0.201 TLoss:0.230 CLoss:1.037 OLoss:0.010 
Epoch: [78][450/695] DCSLoss:0.190 DPLoss:-0.224 Loss:1.024 iLoss:0.283 TLoss:0.152 CLoss:1.002 OLoss:0.010 
Epoch: [78][500/695] DCSLoss:0.198 DPLoss:-0.226 Loss:0.960 iLoss:0.240 TLoss:0.166 CLoss:0.920 OLoss:0.010 
Epoch: [78][550/695] DCSLoss:0.216 DPLoss:-0.225 Loss:1.355 iLoss:0.497 TLoss:0.228 CLoss:1.033 OLoss:0.010 
Epoch: [78][600/695] DCSLoss:0.204 DPLoss:-0.224 Loss:1.044 iLoss:0.243 TLoss:0.207 CLoss:0.988 OLoss:0.010 
Epoch: [78][650/695] DCSLoss:0.233 DPLoss:-0.225 Loss:1.126 iLoss:0.232 TLoss:0.267 CLoss:0.995 OLoss:0.009 
Test Epoch: 78
Extracting Gallery Feature...
Extracting Time:	 0.933
Extracting Query Feature...
Extracting Time:	 12.637
Evaluation Time:	 5.510
POOL:   Rank-1: 76.39% | Rank-5: 93.24% | Rank-10: 97.03%| Rank-20: 99.13%| mAP: 72.17%| mINP: 58.92%
POOL:   Rank-1: 75.94% | Rank-5: 93.22% | Rank-10: 96.74%| Rank-20: 98.97%| mAP: 71.50%| mINP: 57.51%
POOL:   Rank-1: 77.62% | Rank-5: 94.00% | Rank-10: 97.24%| Rank-20: 99.24%| mAP: 73.28%| mINP: 60.03%
Best Epoch [78]

However, when I proceed to test the model using the trained weights with the following command:
 **python test_pdm.py --mode all --tvsearch True --resume 'model_path' --gpu 0 --dataset sysu --protos 10** 
I observe a significant drop in performance, with results around 75% in Rank-1, as detailed below:

==> Building model..
==> Loading data..
==> Resuming from checkpoint..
==> loading checkpoint /home/lunet/coyz2/mlkd_reid/ICASSP2025-PDM/save_model/sysu_deen_p4_n8_lr_0.1_seed_0_best.t
/home/lunet/coyz2/mlkd_reid/ICASSP2025-PDM/test_pdm.py:302: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  checkpoint = torch.load(model_path, map_location='cuda:2')
==> loaded checkpoint /home/lunet/coyz2/mlkd_reid/ICASSP2025-PDM/save_model/sysu_deen_p4_n8_lr_0.1_seed_0_best.t (epoch 78)
Dataset statistics:
  ------------------------------
  subset   | # ids | # images
  ------------------------------
  query    |    96 |     3803
  gallery  |    96 |      301
  ------------------------------
Data Loading Time:       7.497
Extracting Query Feature...
Extracting Time:         27.349
Extracting Gallery Feature...
Extracting Time:         3.072
Test Trial: 0
POOL:   Rank-1: 77.49% | Rank-5: 93.61% | Rank-10: 97.40%| Rank-20: 99.37%| mAP: 74.00%| mINP: 61.80%
POOL:   Rank-1: 77.65% | Rank-5: 93.48% | Rank-10: 97.21%| Rank-20: 99.34%| mAP: 73.65%| mINP: 60.89%
Extracting Gallery Feature...
Extracting Time:         3.846
Test Trial: 1
POOL:   Rank-1: 75.57% | Rank-5: 92.93% | Rank-10: 96.66%| Rank-20: 98.63%| mAP: 73.55%| mINP: 61.54%
POOL:   Rank-1: 75.28% | Rank-5: 92.74% | Rank-10: 96.63%| Rank-20: 98.55%| mAP: 73.09%| mINP: 60.68%
Extracting Gallery Feature...
Extracting Time:         2.112
Test Trial: 2
POOL:   Rank-1: 72.42% | Rank-5: 92.24% | Rank-10: 96.84%| Rank-20: 99.58%| mAP: 70.22%| mINP: 58.49%
POOL:   Rank-1: 71.89% | Rank-5: 91.82% | Rank-10: 96.84%| Rank-20: 99.55%| mAP: 69.68%| mINP: 57.85%
Extracting Gallery Feature...
Extracting Time:         2.089
Test Trial: 3
POOL:   Rank-1: 78.31% | Rank-5: 93.35% | Rank-10: 96.84%| Rank-20: 99.37%| mAP: 73.72%| mINP: 60.61%
POOL:   Rank-1: 77.99% | Rank-5: 93.19% | Rank-10: 97.05%| Rank-20: 99.29%| mAP: 73.36%| mINP: 59.99%
Extracting Gallery Feature...
Extracting Time:         2.061
Test Trial: 4
POOL:   Rank-1: 76.65% | Rank-5: 93.14% | Rank-10: 97.48%| Rank-20: 98.95%| mAP: 74.21%| mINP: 62.95%
POOL:   Rank-1: 76.47% | Rank-5: 93.19% | Rank-10: 97.37%| Rank-20: 98.97%| mAP: 73.89%| mINP: 62.27%
Extracting Gallery Feature...
Extracting Time:         2.027
Test Trial: 5
POOL:   Rank-1: 73.13% | Rank-5: 91.27% | Rank-10: 96.77%| Rank-20: 99.00%| mAP: 71.42%| mINP: 60.40%
POOL:   Rank-1: 73.07% | Rank-5: 91.27% | Rank-10: 96.90%| Rank-20: 98.90%| mAP: 71.14%| mINP: 59.91%
Extracting Gallery Feature...
Extracting Time:         2.008
Test Trial: 6
POOL:   Rank-1: 74.57% | Rank-5: 94.29% | Rank-10: 97.45%| Rank-20: 99.47%| mAP: 71.52%| mINP: 58.40%
POOL:   Rank-1: 74.70% | Rank-5: 94.19% | Rank-10: 97.66%| Rank-20: 99.66%| mAP: 71.20%| mINP: 57.64%
Extracting Gallery Feature...
Extracting Time:         2.027
Test Trial: 7
POOL:   Rank-1: 77.18% | Rank-5: 91.95% | Rank-10: 95.61%| Rank-20: 98.69%| mAP: 72.61%| mINP: 59.79%
POOL:   Rank-1: 76.78% | Rank-5: 91.77% | Rank-10: 95.71%| Rank-20: 98.58%| mAP: 72.31%| mINP: 59.44%
Extracting Gallery Feature...
Extracting Time:         2.088
Test Trial: 8
POOL:   Rank-1: 75.18% | Rank-5: 92.37% | Rank-10: 96.87%| Rank-20: 99.58%| mAP: 71.90%| mINP: 58.86%
POOL:   Rank-1: 75.34% | Rank-5: 92.06% | Rank-10: 96.82%| Rank-20: 99.61%| mAP: 71.49%| mINP: 58.21%
Extracting Gallery Feature...
Extracting Time:         3.731
Test Trial: 9
POOL:   Rank-1: 75.49% | Rank-5: 92.06% | Rank-10: 96.98%| Rank-20: 99.58%| mAP: 72.17%| mINP: 60.24%
POOL:   Rank-1: 75.47% | Rank-5: 92.11% | Rank-10: 97.05%| Rank-20: 99.63%| mAP: 71.95%| mINP: 59.86%
All Average:
POOL:   Rank-1: 75.60% | Rank-5: 92.72% | Rank-10: 96.89%| Rank-20: 99.22%| mAP: 72.53%| mINP: 60.31%
POOL:   Rank-1: 75.46% | Rank-5: 92.58% | Rank-10: 96.93%| Rank-20: 99.21%| mAP: 72.18%| mINP: 66.29%

The results are considerably lower than the training epoch evaluation and what reported in the paper.

Could you kindly provide any insights or potential reasons for this discrepancy? Any guidance would be greatly appreciated.
